framework/auth/core.py

```diff
@@ -1126,6 +1126,11 @@ def get_daily_digest_log_ids(self):
         )
         return self.get_recent_log_ids(since=midnight)
 
+    def files_checked_out(self):
```

Comment:
We try to avoid mixing too much file and node-related logic with our User model. It might make more sense to move this to a `FileNode` method.

```python
FileNode.find_checked_out_by_user(user)
```

That way, we localize much of the node/file/user mixing to the `merge_user` method.